### PR TITLE
[fix](scanner) Fix incorrect _max_thread_num in scanner context

### PR DESCRIPTION
**Pull request by zhiqiang-hhhh**
1. Minor refactor for scanner constructor, calculation of _max_thread_num is moved to init method
2. The expected value of _max_thread_num is changed. There is no need to submit too many scan task to scan scheduler, since thread num is limited.
3. Calculation of _max_bytes_in_queue is changed. _max_bytes_in_queue for each scan instance is limited to 100MB by default.

### CODE REVIEW

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: be/src/vec/exec/scan/scanner_context.cpp [hunk 1]
```diff
@@ -53,53 +52,102 @@ ScannerContext::ScannerContext(
...
+    // https://github.com/apache/doris/pull/28266
+    if (ignore_data_distribution) {
+        num_parallel_instances = 1;
+    }
+
+    // _max_bytes_in_queue controls the maximum memory that can be used by a single scan instance.
+    // scan_queue_mem_limit on FE is 100MB by default, on backend we will make sure its actual value
+    // is larger than 10MB.
+    _max_bytes_in_queue =
+            std::max(_state->scan_queue_mem_limit(), (int64_t)1024 * 1024 * 1024 * 10);
```
Comment (yiguolei) [inline be/src/vec/exec/scan/scanner_context.cpp:109]: 是10MB， 不是10gb

Comment (zhiqiang-hhhh) [reply be/src/vec/exec/scan/scanner_context.cpp]: fixed

File: be/src/pipeline/exec/scan_operator.cpp [hunk 2]
```diff
@@ -117,7 +117,7 @@ Status ScanLocalState<Derived>::open(RuntimeState* state) {
     RETURN_IF_ERROR(status);
     if (_scanner_ctx) {
         DCHECK(!_eos && _num_scanners->value() > 0);
-        RETURN_IF_ERROR(_scanner_ctx->init());
+        RETURN_IF_ERROR(_scanner_ctx->init(p.ignore_data_distribution()));
```
Comment (yiguolei) [inline be/src/pipeline/exec/scan_operator.cpp:120]: 可能把，这个参数，作为scanner context的构造函数的参数，传递给scanner context，这样你往2.1 pick的时候会比较方便。否则2.1 在多种pipeline 模型下，代码不好pick

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Comment by yiguolei** [comment]
run performance

**Review by yiguolei — APPROVED** [review]

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**